app/controllers/time_slot_controller.rb

```diff
@@ -39,12 +45,11 @@ def index
     # FIXME We need to change this relation to has_one or enhance the logic here
     @appointment = current_patient.appointments.last
     @ubs = @appointment.try(:ubs)
+    
+    date_range = build_weekdays_date_range(TimeSlotController::SLOTS_WINDOW_IN_DAYS)
 
     @time_slots = Ubs.all.where(active: true).each_with_object({}) do |ubs, memo|
-      memo[ubs] = ubs.available_time_slots(Time.zone.today...3.days.from_now, Time.zone.now)
-
-      # TODO: Refactor to not include these as available
-      memo.delete(ubs) if memo[ubs].empty?
+      memo[ubs] = ubs.available_time_slots(date_range, Time.zone.now)
```

Comment:
@laurogripa including this suggested changes, I think now this part is more clean and readable.